src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -920,8 +920,17 @@ int32_t SystemNative_Poll(PollEvent* pollEvents, uint32_t eventCount, int32_t mi
     }
 
 
-    int useStackBuffer = bufferSize <= 2048;
-    struct pollfd* pollfds = (struct pollfd*)(useStackBuffer ? alloca(bufferSize) : malloc(bufferSize));
+    char stackBuffer[2048];
+    int useStackBuffer = bufferSize <= sizeof(stackBuffer);
+    struct pollfd* pollfds;
+    if (useStackBuffer)
+    {
+        pollfds = (struct pollfd*)(&stackBuffer[0]);
+    }
+    else
+    {
+        pollfds = (struct pollfd*)(malloc(bufferSize));
+    }
     if (pollfds == NULL)
     {
         return Error_ENOMEM;
```

Comment:
Also, unrelated to this commit: the events in `poll()` is a bitmask. The conversion to/from the PAL seems wrong (it's probably fine if nobody ever asks for something like "I want Input and Priority events".)